From: Yevgeny Kliteynik <kliteyn@nvidia.com>
Subject: [PATCH] BACKPORT:
 drivers/net/ethernet/mellanox/mlx5/core/steering/hws/bwc_complex.c

Change-Id: Ica54b6a62fd97072179a1fabb3f7f4ad77928f7d
---
 .../mellanox/mlx5/core/steering/hws/bwc_complex.c    | 12 ++++++++++++
 1 file changed, 12 insertions(+)

--- a/drivers/net/ethernet/mellanox/mlx5/core/steering/hws/bwc_complex.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/steering/hws/bwc_complex.c
@@ -675,7 +675,11 @@ __must_hold(&subm->hash_lock)
 	if (!sr_data)
 		return -ENOMEM;
 
+#ifdef HAVE_IDA_ALLOC
 	ret = ida_alloc(&subm->chain_ida, GFP_KERNEL);
+#else
+	ret = ida_simple_get(&subm->chain_ida, 0, 0, GFP_KERNEL);
+#endif
 	if (ret < 0)
 		goto free_sr_data;
 	sr_data->chain_id = ret;
@@ -714,7 +718,11 @@ __must_hold(&subm->hash_lock)
 	return 0;
 
 free_ida:
+#ifdef HAVE_IDA_ALLOC
 	ida_free(&subm->chain_ida, sr_data->chain_id);
+#else
+	ida_simple_remove(&subm->chain_ida, sr_data->chain_id);
+#endif
 free_sr_data:
 	kfree(sr_data);
 
@@ -737,7 +745,11 @@ __must_hold(&subm->hash_lock)
 		rhashtable_remove_fast(&subm->rules_hash,
 				       &sr_data->hash_node,
 				       hws_rules_hash_params);
+#ifdef HAVE_IDA_ALLOC
 		ida_free(&subm->chain_ida, sr_data->chain_id);
+#else
+		ida_simple_remove(&subm->chain_ida, sr_data->chain_id);
+#endif
 		kfree(sr_data);
 		if (is_last_rule)
 			*is_last_rule = true;
